platforms/core-runtime/base-services/src/main/java/org/gradle/internal/service/DefaultServiceRegistry.java

```diff
@@ -90,11 +90,11 @@ private enum State {INIT, STARTED, CLOSED}
 
     private final ClassInspector inspector;
     private final OwnServices ownServices;
-    private final ServiceProvider allServices;
-    private final ServiceProvider parentServices;
+    private final ServiceSource allServices;
+    private final ServiceSource parentServices;
```

Comment:
💭 I've renamed this internal `ServiceProvider` type to `ServiceSource` to free up the `ServiceProvider` name for classes that use `@Provides` methods to provide services.

Reply:
But those are real service providers, they provide services, and the new marker interface is about providing _service registrations_.